include/baresip.h

```diff
@@ -827,6 +827,8 @@ enum ua_event {
 	UA_EVENT_CALL_MENC,
 	UA_EVENT_VU_TX,
 	UA_EVENT_VU_RX,
+	UA_EVENT_VAD_TX,
+	UA_EVENT_VAD_RX,
```

Comment:
I suggest to
- Use `UA_EVENT_MODULE` with specific parameter for vad.
- Use function `module_event()` in the module `fvad`. It is not necessary that the event is generated in `call.c`.

It should be possible to implement everything in the module, without any changes in the baresip core. @alfredh should this go to baresip-apps?

The application could evaluate the specific module event.

Reply:
Ah, ok. I'll try to rewrite it using module events.

Reply:
When sending a module event, I need to pass a `struct call *` or a `struct ua *` to `module_event` so I can make sure the right ua/call processes the event, but all I get is a `struct audio *` and I can't see how to get back at the call from there. Any ideas?

Reply:
Yes. Currently this is not directly possible. But the list of calls can be searched for the right audio stream. Use `uag_filter_calls()` and pass your list and match handlers!

Reply:
Ok, in this case, the `module_event` function would have to get an optional `struct audio *` (and a `struct video *` too) so I can filter in the app.

Or a single `void *` pointer is passed and an informational enum, what sort of pointer it is. Something like:

```
enum ptr_type {
    PTR_CALL,
    PTR_UA,
    PTR_AUDIO,
    PTR_VIDEO
};

void module_event(const char *module, const char *event, ptr_type t, void *sender,
		const char *fmt, ...);
```
I don't like the idea with the back pointer much either.

Reply:
`encode_update()` and `decode_update()` has a pointer to audio. `uag_filter_calls()` only needs the handlers that your module could provide.